scuttlebutt-handshake/src/main/java/net/consensys/cava/scuttlebutt/handshake/vertx/SecureScuttlebuttVertxServer.java

```diff
@@ -79,10 +81,34 @@ private void handleMessage(Buffer buffer) {
           });
         } else {
           Bytes message = streamServer.readFromClient(Bytes.wrapBuffer(buffer));
-          if (SecureScuttlebuttStreamServer.isGoodbye(message)) {
-            netSocket.close();
-          } else {
-            handler.receivedMessage(message);
+          messageBuffer = Bytes.concatenate(messageBuffer, message);
+
+          // Process any whole RPC message repsonses we have, and leave any partial ones at the end in the buffer
```

Comment:
Just a thought in passing - this might be decoupled down the road, as it is possible to use SSB to transmit data without using RPC.